libraries/AP_AS5600_AOA/AS5600_AOA.cpp

```diff
@@ -0,0 +1,266 @@
+void AS5600_AOA::setOutPut(unsigned char mode){
+    unsigned char config_status;
+    config_status = readOneByte(_conf_lo);
+    if(mode == 1){
+        config_status = config_status & 0xcf;
+    }else{
+        config_status = config_status & 0xef;
+    }
+
+    /* Note significant variance from the Arduino AMS_5600 library in this line.
+     *
+     * -> writeOneByte(_conf_lo, lowByte(config_status));
+     *  vs writeOneByte(_conf_lo, config_status); I have removed the lowByte() function
+     *  since it is part of the arduino.h library and I don't have access to it, however,
+     *  since it accesses the lowest byte in the variable, and an unsigned char such as
+     *  config_status only HAS one byte regardless, I think the end result should be identical?
+     */
+    writeOneByte(_conf_lo, config_status);
+}
+
+
+/*******************************************************
+ * Method: getRawAngle
+ * In: none
+ * Out: value of raw angle register
+ * Description: gets raw value of magnet position.
+ * start, end, and max angle settings do not apply
+*******************************************************/
+unsigned short AS5600_AOA::getRawAngle(void)
+{
+  unsigned short angleRaw = readTwoBytes(_raw_ang_hi, _raw_ang_lo);
+
+  unsigned short angleDegrees = angleRaw*0.087;
+
+  AP::logger().Write("AoAR", "Status, TimeUS, Angle", "iQH", int(bool(dev)), AP_HAL::micros64(), angleDegrees);
+
+   return angleDegrees;
+}
+/*******************************************************
+ * Method: highByte
+ * In: Unsigned short
+ * Out: Highest or leftmost byte of the unsigned short
+ * Description: Takes in the unsigned short and returns
+ * the leftmost or highest bite
+*******************************************************/
+
+unsigned char AS5600_AOA::highByte(unsigned short short_in){
+
+    unsigned char hiByte = ((short_in >> 8) & 0xff);
+    return  hiByte;
+}
+
+/*******************************************************
+ * Method: lowByte
+ * In: Unsigned short
+ * Out: Lowest or rightmost byte of the unsigned short
+ * Description: Takes in the unsigned short and returns
+ * the rightmost or lowest byte
+*******************************************************/
+
+unsigned char AS5600_AOA::lowByte(unsigned short short_in){
+
+    unsigned char loByte = (short_in & 0xff);
+    return  loByte;
+}
+
+
+/*******************************************************
+ * Method: setMaxAngle
+ * In: new maximum angle to set OR none
+ * Out: value of max angle register
+ * Description: sets a value in maximum angle register.
+ * If no value is provided, method will read position of
+ * magnet.  Setting this register zeros out max position
+ * register.
+*******************************************************/
+/*unsigned short AS5600_AOA::setMaxAngle(unsigned short newMaxAngle)
+{
+  unsigned short retVal;
+  if(newMaxAngle == -1)
+  {
+    maxAngle = getRawAngle();
+  }
+  else
+    maxAngle = newMaxAngle;

```

Comment:
```suggestion
  if(newMaxAngle == -1){
    maxAngle = getRawAngle();
  } else {
     maxAngle = newMaxAngle;
  }
```

This is commented out, so no more review on this for the time being.